### PR TITLE
Add a common terminology glossary page

### LINKED ISSUE #198: Add a common terminology glossary page
There should be a description of deprecated somewhere. Most high school students don't know what it means, and when they see warnings they think their code is broken.

### PR DESCRIPTION
**Pull request by jasondaming**
Closes #198 

I need more ideas of things that need to be included!

### CODE REVIEW

File: source/docs/software/frc-glossary.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+   KOP chassis
+      The `AM14U4 <https://www.andymark.com/products/am14u4-kit-of-parts-chassis>`__ chassis distributed to every team in the :term:`KOP`.
+
+   LabVIEW
+      One of the three officially supported programming languages.
+
+   match
+      A two (2) minute and thirty (30) second period of time in which an :term:`alliance` plays a FRC game.
+
+   NT
```
Comment (Daltz333) [inline source/docs/software/frc-glossary.rst:59]: > NT
Not really anywhere this is referenced as NT. Just leave it as NetworkTables

File: source/docs/software/frc-glossary.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+   auto
+      The first phase of each :term:`match` is called Autonomous (auto) and consists of the first fifteen (0:15) seconds.
+
+   COTS
+      Commercial off the shelf, a standard (i.e. not custom order) part commonly available from a vendor to all teams for purchase.
+
+   C++
+      One of the three officially supported programming languages.
+
+   depricated
```
Comment (Daltz333) [inline source/docs/software/frc-glossary.rst:23]:
```suggestion
   deprecated
```

File: source/index.rst [hunk 3]
```diff
@@ -33,6 +33,7 @@ Welcome to the *FIRST*\ |reg| Robotics Competition Control System Documentation!
    :caption: Programming Basics
 
    docs/software/what-is-wpilib
+   docs/software/frc-glossary
```
Comment (Daltz333) [inline source/index.rst:36]: Thinking about it, maybe we should move this low down in the index. I don't think we want people to actively read it, but more use it as a page to reference for ``:term:``

File: source/docs/software/frc-glossary.rst (hunk 1 above)
Comment (jasondaming) [reply source/docs/software/frc-glossary.rst:59]: I don't think it is as rare as you think, from our docs:
Twice [here](https://docs.wpilib.org/en/latest/docs/software/vision-processing/roborio/using-multiple-cameras.html?highlight=NT#switching-the-driver-views)
Once [here](https://docs.wpilib.org/en/latest/docs/software/wpilib-tools/shuffleboard/custom-widgets/builtin-plugins.html?highlight=NT#networktables-plugin)
The rPi abbreviates is this way in the logs [here](https://docs.wpilib.org/en/latest/docs/software/vision-processing/wpilibpi/the-raspberry-pi-frc-console.html?highlight=NT#vision-status)
[Here](https://docs.wpilib.org/en/latest/docs/software/networktables/multiple-instances.html?highlight=NT#creating-multiple-instances-of-networktables)

There are probably more.

File: source/index.rst (hunk 3 above)
Comment (jasondaming) [reply source/index.rst:36]: makes sense

File: source/docs/software/frc-glossary.rst (hunk 1 above)
Comment (Daltz333) [reply source/docs/software/frc-glossary.rst:59]: I think in this specific case we should prefer NetworkTables, we can still reference to it when doing NT by doing ```:term:`NT <NetworkTables>` ``` 

We should probably avoid two letter abbreviations as they can quickly collide.

File: source/docs/software/frc-glossary.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
+.. include:: <isonum.txt>
+
+FRC Glossary
+============
+
+.. glossary::
+
+   accelerometer
+      A common sensors used to measure acceleration in one or more axis.
```
Comment (prateekma) [inline source/docs/software/frc-glossary.rst:9]: sensors -> sensor

File: source/docs/software/frc-glossary.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      Dynamic Host Configuration Protocol, the protocol that allows a central device to assign unique IP addresses to all other devices.
+
+   FMS
+      Field Management System, the electronics core responsible for sensing and controlling the FIRST Robotics Competition field.
+
+   GradleRIO
+      The mechanism that powers the deployment of robot code to the roboRIO.
+
+   gyroscope
+      A device that measures rate-of-rotation.  It can add up the rotation measurements to determine :term:`heading` of the robot. (“gyro”, for short)
```
Comment (prateekma) [inline source/docs/software/frc-glossary.rst:36]: I don't think the hyphens are necessary here. Just "rate of rotation" should work.

File: source/docs/software/frc-glossary.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      Inertial Measurement Unit, a sensor that combines both an :term:`accelerometer` and a :term:`gyroscope` into a single sensor.
+
+   Java
+      One of the three officially supported programming languages.
+
+   KOP
+      Kit of Parts, the collection of items listed on the Kickoff Kit checklists, distributed to the team via FIRST Choice, or paid for completely (except shipping) with a Product Donation Voucher (PDV).
+
+   KOP chassis
+      The `AM14U4 <https://www.andymark.com/products/am14u4-kit-of-parts-chassis>`__ chassis distributed to every team in the :term:`KOP`.
```
Comment (prateekma) [inline source/docs/software/frc-glossary.rst:51]: Should we mention that this is opt-out?

**Review by prateekma — CHANGES_REQUESTED** [review]

**Review by prateekma — APPROVED** [review]

**Comment by Daltz333** [comment]
Can we add a note somewhere in the styleguide that common used terms should get added here?

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**